src/Plugins/ApplicationLogs/Store/LogStorageStore.cs

```diff
@@ -159,9 +160,13 @@ public Guid PutStackItemState(StackItem stackItem)
             {
                 _snapshot.Put(key, BinarySerializer.Serialize(stackItem, ExecutionEngineLimits.Default with { MaxItemSize = (uint)Settings.Default.MaxStackSize }));
             }
-            catch (NotSupportedException)
+            catch (Exception ex)
             {
-                _snapshot.Put(key, BinarySerializer.Serialize(StackItem.Null, ExecutionEngineLimits.Default with { MaxItemSize = (uint)Settings.Default.MaxStackSize }));
+                var detailedException = ex.Message;
+                if (detailedException.Length > 1024)
```

Comment:
Thanks @Jim8y. You've create a lot of PRs that makes exception messages much more clear than before. So as this one.

It would be better if we change the magic 1024 to some constant variable like `Default.MaxStackSize/2` or something else.

Reply:
@dusmart Thank you for the recognition, credits of this fix goes to @Hecate2. Need @dusmart your suggestion to decide that value, is Default.MaxStackSize/2 secure enough.

Reply:
I'm not sure what's the real value of `Default.MaxStackSize`. 1024 and 2048 or any number, they all seems good to me. Maybe @AnnaShaleva has more experience on this settings.

Reply:
@dusmart, to be honest, I'd prefer not mix the real (even invalid) stackitem with some system-created one. See the https://github.com/neo-project/neo/pull/3299#discussion_r1630873140.

Reply:
> I'm not sure what's the real value of `Default.MaxStackSize`. 1024 and 2048 or any number, they all seems good to me. Maybe @AnnaShaleva has more experience on this settings.

Settings comes the `config.json` so it can be whatever you want.